From 1820c534b9a15008b9c3879d31c2b144f1c6de49 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Fri, 17 May 2024 13:33:17 +0300
Subject: [PATCH 084/155] compulab: iotdin-imx8p: update device tree and
 defconfig

Sync device tree with ucm-imx8m-plus.
Update defconfig:
 - disable VIDEO support
 - force bsp_bootcmd as a default boot command

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 arch/arm/dts/iotdin-imx8p-u-boot.dtsi | 114 +++++++++++++++++---------
 arch/arm/dts/sb-iotdimx8p.dtsi        |   8 +-
 configs/iotdin-imx8p_defconfig        |  14 +---
 3 files changed, 80 insertions(+), 56 deletions(-)

diff --git a/arch/arm/dts/iotdin-imx8p-u-boot.dtsi b/arch/arm/dts/iotdin-imx8p-u-boot.dtsi
index 4e7766c531a..08b2e34e4fc 100644
--- a/arch/arm/dts/iotdin-imx8p-u-boot.dtsi
+++ b/arch/arm/dts/iotdin-imx8p-u-boot.dtsi
@@ -1,6 +1,5 @@
 // SPDX-License-Identifier: GPL-2.0+
 /*
- * Copyright 2019 NXP
  * Copyright 2024 CompuLab
  */
 
@@ -21,86 +20,108 @@
 			method = "smc";
 		};
 	};
-};
-
-&{/soc@0} {
-	u-boot,dm-pre-reloc;
-	u-boot,dm-spl;
-};
-
-&clk {
-	u-boot,dm-spl;
-	u-boot,dm-pre-reloc;
-	/delete-property/ assigned-clocks;
-	/delete-property/ assigned-clock-parents;
-	/delete-property/ assigned-clock-rates;
-
-};
 
-&osc_32k {
-	u-boot,dm-spl;
-	u-boot,dm-pre-reloc;
-};
-
-&osc_24m {
-	u-boot,dm-spl;
-	u-boot,dm-pre-reloc;
+	mcu_rdc {
+		compatible = "imx8m,mcu_rdc";
+			    /* rdc config when MCU starts
+			     * master
+			     *   SDMA3p --> domain 1
+			     *   SDMA3b --> domian 1
+			     *   SDMA3_SPBA2  --> domian 1
+			     * peripheral:
+			     *   SAI3   --> Only Domian 1 can access
+			     *   UART4  --> Only Domian 1 can access
+			     *   GPT1   --> Only Domian 1 can access
+			     *   SDMA3  --> Only Domian 1 can access
+			     *   I2C3   --> Only Domian 1 can access
+			     * memory:
+			     *   TCM    --> Only Domian 1 can access (0x7E0000~0x81FFFF)
+			     *   DDR    --> Only Domian 1 can access (0x80000000~0x81000000)
+			     * end.
+			     */
+		start-config = <
+			    RDC_MDA  RDC_MDA_SDMA3p DID1 0x0 0x0
+			    RDC_MDA  RDC_MDA_ENET1_TX DID1 0x0 0x0
+			    RDC_MDA  RDC_MDA_ENET1_RX DID1 0x0 0x0
+			    RDC_MDA  RDC_MDA_SDMA3b DID1 0x0 0x0
+			    RDC_MDA  RDC_MDA_SDMA3_SPBA2 DID1 0x0 0x0
+			    RDC_PDAP RDC_PDAP_ENET1 PDAP_D0D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_SAI3  PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_UART4 PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_GPT1  PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_SDMA3 PDAP_D1_ACCESS 0x0 0x0
+			    RDC_PDAP RDC_PDAP_I2C3  PDAP_D1_ACCESS 0x0 0x0
+			    RDC_MEM_REGION 22 TCM_START TCM_END MEM_D1_ACCESS
+			    RDC_MEM_REGION 39 M4_DDR_START M4_DDR_END MEM_D1_ACCESS
+			    0x0      0x0            0x0  0x0 0x0
+			  >;
+			    /* rdc config when MCU stops
+			     * memory:
+			     *   TCM    --> domain 0/1 can access (0x7E0000~0x81FFFF)
+			     *   DDR    --> domain 0/1 can access (0x80000000~0x81000000)
+			     * end.
+			     */
+		stop-config = <
+			    RDC_MEM_REGION 22 TCM_START TCM_END MEM_D0D1_ACCESS
+			    RDC_MEM_REGION 39 M4_DDR_START M4_DDR_END MEM_D0D1_ACCESS
+			    0x0      0x0            0x0  0x0 0x0
+			  >;
+	};
 };
 
-&aips1 {
+&pinctrl_usb_hub_rst {
 	u-boot,dm-spl;
-	u-boot,dm-pre-reloc;
 };
 
-&aips2 {
+&pinctrl_usb2_vbus {
 	u-boot,dm-spl;
 };
 
-&aips3 {
+&pinctrl_uart2 {
 	u-boot,dm-spl;
 };
 
-&iomuxc {
+&pinctrl_usdhc3 {
 	u-boot,dm-spl;
 };
 
-&pinctrl_usb_hub_rst {
+&gpio1 {
 	u-boot,dm-spl;
 };
 
-&pinctrl_usb2_vbus {
+&gpio2 {
 	u-boot,dm-spl;
 };
 
-&pinctrl_uart2 {
+&gpio3 {
 	u-boot,dm-spl;
 };
 
-&pinctrl_usdhc3 {
+&gpio4 {
 	u-boot,dm-spl;
 };
 
-&gpio1 {
+&gpio5 {
 	u-boot,dm-spl;
 };
 
-&gpio2 {
+&uart2 {
 	u-boot,dm-spl;
 };
 
-&gpio3 {
+&crypto {
 	u-boot,dm-spl;
 };
 
-&gpio4 {
+&sec_jr0 {
 	u-boot,dm-spl;
 };
 
-&gpio5 {
+&sec_jr1 {
 	u-boot,dm-spl;
 };
 
-&uart2 {
+&sec_jr2 {
 	u-boot,dm-spl;
 };
 
@@ -174,8 +195,21 @@
 };
 
 &eqos {
-	compatible = "fsl,imx-eqos";
 	/delete-property/ assigned-clocks;
 	/delete-property/ assigned-clock-parents;
 	/delete-property/ assigned-clock-rates;
 };
+
+&usb_dwc3_0 {
+	compatible = "fsl,imx8mq-dwc3", "snps,dwc3";
+	assigned-clocks = <&clk IMX8MP_CLK_HSIO_AXI>;
+	assigned-clock-parents = <&clk IMX8MP_SYS_PLL1_800M>;
+	assigned-clock-rates = <400000000>;
+};
+
+&usb_dwc3_1 {
+	compatible = "fsl,imx8mq-dwc3", "snps,dwc3";
+	assigned-clocks = <&clk IMX8MP_CLK_HSIO_AXI>;
+	assigned-clock-parents = <&clk IMX8MP_SYS_PLL1_800M>;
+	assigned-clock-rates = <400000000>;
+};
diff --git a/arch/arm/dts/sb-iotdimx8p.dtsi b/arch/arm/dts/sb-iotdimx8p.dtsi
index 4212af12fc9..ed107cbbf34 100644
--- a/arch/arm/dts/sb-iotdimx8p.dtsi
+++ b/arch/arm/dts/sb-iotdimx8p.dtsi
@@ -34,9 +34,7 @@
 	pinctrl-0 = <&pinctrl_fec>, <&pinctrl_fec_phy_reset>;
 	phy-mode = "rgmii-id";
 	phy-handle = <&fecphy>;
-	phy-reset-gpios = <&gpio4 24 GPIO_ACTIVE_LOW>;
-	phy-reset-duration = <10>;
-	phy-reset-post-delay = <150>;
+	fsl,magic-packet;
 	status = "okay";
 
 	mdio {
@@ -48,6 +46,10 @@
 			compatible = "ethernet-phy-ieee802.3-c22";
 			reg = <0>;
 			eee-broken-1000t;
+			reset-gpios = <&gpio4 24 GPIO_ACTIVE_LOW>;
+			reset-assert-us = <15000>;
+			reset-deassert-us = <100000>;
+			realtek,clkout-disable;
 		};
 	};
 };
diff --git a/configs/iotdin-imx8p_defconfig b/configs/iotdin-imx8p_defconfig
index b902f3ecb87..4f5e89b9779 100644
--- a/configs/iotdin-imx8p_defconfig
+++ b/configs/iotdin-imx8p_defconfig
@@ -38,7 +38,7 @@ CONFIG_LEGACY_IMAGE_FORMAT=y
 CONFIG_OF_BOARD_SETUP=y
 CONFIG_OF_SYSTEM_SETUP=y
 CONFIG_DISTRO_DEFAULTS=y
-CONFIG_BOOTCOMMAND="run distro_bootcmd;run bsp_bootcmd"
+CONFIG_BOOTCOMMAND="run bsp_bootcmd"
 CONFIG_ARCH_MISC_INIT=y
 CONFIG_BOARD_EARLY_INIT_F=y
 CONFIG_BOARD_LATE_INIT=y
@@ -174,18 +174,6 @@ CONFIG_USB_GADGET=y
 CONFIG_USB_GADGET_MANUFACTURER="FSL"
 CONFIG_USB_GADGET_VENDOR_NUM=0x1fc9
 CONFIG_USB_GADGET_PRODUCT_NUM=0x0152
-CONFIG_VIDEO=y
-CONFIG_VIDEO_LOGO=y
-CONFIG_SYS_WHITE_ON_BLACK=y
-CONFIG_VIDEO_LCD_STARTEK_ILI9881C=y
-CONFIG_VIDEO_IMX8MP_LVDS=y
-CONFIG_VIDEO_IMX_SEC_DSI=y
-CONFIG_VIDEO_IMX_LCDIFV3=y
-CONFIG_SPLASH_SCREEN=y
-CONFIG_SPLASH_SCREEN_ALIGN=y
-CONFIG_BMP_16BPP=y
-CONFIG_BMP_24BPP=y
-CONFIG_BMP_32BPP=y
 CONFIG_WDT=y
 CONFIG_LZO=y
 CONFIG_BZIP2=y
-- 
2.34.1

